tools/dockerfile/distribtest/python_dev_centos7_x64/Dockerfile

```diff
@@ -18,5 +18,13 @@ RUN yum install -y python
 RUN yum install -y epel-release
 RUN yum install -y python-pip
 RUN pip install virtualenv
-RUN yum groupinstall -y 'Development Tools'
 RUN yum install -y python-devel
+
+# The default gcc of CentOS 7 is gcc 4.8 which is older than gcc 4.9,
+# the minimum supported gcc version for gRPC Core so let's upgrade to
+# the latest one available on Centos 7.
+RUN yum install -y centos-release-scl
+RUN yum install -y devtoolset-8-binutils devtoolset-8-gcc devtoolset-8-gcc-c++
```

Comment:
If gcc4.8 is not supported, why is the test not currently failing then? Or are we just proactively trying to avoid potential issues?

Would it make more sense to use an older gcc version for the purpose of a distribtest? Generally distribtests are here so that we don't break compatibility with older stuff by e.g. modernizing our codebase. 
From that perspective, jumping from gcc4.8 to gcc8 in this PR might not make that much sense. Leaving the resolution up to you, just wanted to point this out.

Reply:
This is required by #23723 which is PR to upgrade the Abseil to the upcoming LTS release. gRPC has been built with gcc4.8 but the unit tests hasn't and new Abseil won't be built with gcc 4.8 anymore.

For the version of gcc here, 8 is actually the oldest one which are available on CentOS 7. There are three versions; 7, 8, and 9 and gcc 7 has a compiler bug which cannot build Abseil properly so 8 is chosen.

Reply:
sg, thanks for the explantion.